lib/omise/resource.rb

```diff
@@ -31,23 +36,30 @@ def initialize(url, path, key)
     attr_reader :uri, :headers, :key
 
     def get(attributes = {})
-      @resource.get(params: attributes) { |r| Omise::Util.load_response(r) }
+      resource.get(params: attributes) { |r| load_response(r) }
     end
 
     def patch(attributes = {})
-      @resource.patch(attributes) { |r| Omise::Util.load_response(r) }
+      resource.patch(attributes) { |r| load_response(r) }
```

Comment:
I think that the second argument inside of the block is the `RestClient::Request` so you could turn it into this:

```ruby
resource.patch(attributes) do |response,request|
  log_request(request)
  load_and_log_response(response)
end
```

And you may be able to forgo `RestClient.add_before_execution_proc` completely.